sklearn/tree/_tree.pxd

```diff
@@ -43,6 +46,10 @@ cdef class Tree:
     cdef SIZE_t* n_classes               # Number of classes in y[:, k]
     cdef public SIZE_t n_outputs         # Number of outputs in y
     cdef public SIZE_t max_n_classes     # max(n_classes)
+    cdef object random_state             # Random state
+    cdef UINT32_t rand_r_state           # sklearn_rand_r random number state
```

Comment:
Why do we need these two references? In particular, I am not sure pickling the random_state object is a good idea.